mxresolv/mxresolv.go

```diff
@@ -39,54 +39,78 @@ func init() {
 	lookupResultCache = collections.NewLRUCache(cacheSize)
 }
 
-// Lookup performs a DNS lookup of MX records for the specified hostname. It
+// Lookup performs a DNS lookup of MX records for the specified domain. It
 // returns a prioritised list of MX hostnames, where hostnames with the same
 // priority are shuffled. If the second returned value is true, then the host
 // does not have explicit MX records, and its A record is returned instead.
 //
 // It uses an LRU cache with a timeout to reduce the number of network requests.
-func Lookup(ctx context.Context, hostname string) (retMxHosts []string, retImplicit bool, reterr error) {
-	if cachedVal, ok := lookupResultCache.Get(hostname); ok {
+func Lookup(ctx context.Context, domain string) ([]string, bool, error) {
```

Comment:
I agree with the linter here. The names help with code completion for quick-glance reading